chunky/src/java/se/llbit/chunky/block/legacy/LegacyBlocks.java

```diff
@@ -9,12 +9,27 @@
 
 public class LegacyBlocks {
 
+  /** statically initialised array of every possible tag */
```

Comment:
I'm pretty sure that there are tags that contain additional data in 1.12 worlds, e.g. for bed colors.

Reply:
bed colours work fine with this, just tested it

Reply:
My understanding of the 1.12 saving is that everything is `blockID`(8bits) `meta`(4bits), and we have a function that maps from that to post flattening tags, which is `LegacyBlocks#getTag(int, int)` in this PR

Reply:
If any blocks are broken now, they would've also been broken before this PR

Reply:
You're right… additional data is done by using block entities, e.g. for beds: https://github.com/chunky-dev/chunky/blob/master/chunky/src/java/se/llbit/chunky/block/legacy/blocks/LegacyBed.java#L21, so we can actually use this map to speed up world loading.

How much memory does this map consume? If it's too much, we could build it before loading chunks _once_ so that it could be garbage-collected after loading.

Reply:
Or, we could use some weak map :thinking:

Reply:
Which map are you referring to?